libs/wingsdk/src/target-tf-gcp/table.ts

```diff
@@ -0,0 +1,102 @@
+import { Construct } from "constructs";
+import { App } from "./app";
+import {
+  BigtableInstance,
+  BigtableInstanceCluster,
+  BigtableInstanceClusterAutoscalingConfig,
+  BigtableInstanceConfig,
+} from "../.gen/providers/google/bigtable-instance";
+import {
+  BigtableTable,
+  BigtableTableConfig,
+  BigtableTableColumnFamily,
+} from "../.gen/providers/google/bigtable-table";
+import * as ex from "../ex";
+import {
+  ResourceNames,
+  NameOptions,
+  CaseConventions,
+} from "../shared/resource-names";
+import { IInflightHost, Json } from "../std";
+
+const TABLE_NAME_OPTS: NameOptions = {
+  maxLen: 22,
+  disallowedRegex: /[a-z0-9\-\.\_]+/g,
+  sep: "A",
+};
+
+const INSTANCE_NAME_OPTS: NameOptions = {
+  maxLen: 22,
+  disallowedRegex: /[a-z0-9\-\.\_]+/g,
+  sep: "A",
+  case: CaseConventions.LOWERCASE,
+};
+
+/**
+ * GCP implementation of `ex.Table`.
+ *
+ * @inflight `@winglang/sdk.ex.ITableClient`
+ */
+export class Table extends ex.Table {
+  constructor(scope: Construct, id: string, props: ex.TableProps = {}) {
+    super(scope, id, props);
+
+    const app = App.of(this) as App;
+
+    const tableName = ResourceNames.generateName(this, TABLE_NAME_OPTS);
+    const instanceName = ResourceNames.generateName(this, INSTANCE_NAME_OPTS);
+
+    const columnsFamily: BigtableTableColumnFamily[] = [];
+    for (let key in this.columns) {
+      columnsFamily.push({ family: key });
+    }
+
+    let clusterId = props.clusterId;
+    if (!props.clusterId) {
+      clusterId = "default";
+    }
+
+    const autoscalingConfig: BigtableInstanceClusterAutoscalingConfig = {
+      minNodes: props.minNodes ?? 1,
+      maxNodes: props.maxNodes ?? 1,
+      cpuTarget: props.cpuTarget ?? 10,
+    };
+
+    const instanceCluster: BigtableInstanceCluster = {
+      clusterId: clusterId!,
+      storageType: props.storageType,
+      zone: app.zone,
+      autoscalingConfig: autoscalingConfig,
+    };
+
+    const instanceConfig: BigtableInstanceConfig = {
+      name: instanceName,
+      cluster: [instanceCluster],
+    };
+
+    let instance = new BigtableInstance(this, "Instance", instanceConfig);
+
+    const tableConfig: BigtableTableConfig = {
+      name: tableName,
+      instanceName: instance.name,
+      columnFamily: columnsFamily,
+      project: app.projectId,
+    };
+
+    new BigtableTable(this, "Default", tableConfig);
+  }
+
+  public addRow(_key: string, _row: Json): void {
+    throw new Error("Method is not supported as preflight for GCP target.");
```

Comment:
Google Cloud does not support adding a row without a client, therefore it is not possible with terraform at this stage. I believe it is not possible to implement this as a preflight method - only as an inflight.  Can someone tell me what I should do in this case?

Reply:
I think throwing an error here is ok. 👍  In the future, if they add support in Terraform we can update the implementation. I think occasional gaps like this is normal since the cloud is still new.

Once we have support for the inflight methods of `Table`, we can work around this by making `addRow` use [OnDeploy](https://www.winglang.io/docs/standard-library/cloud/on-deploy) to create a table row at deployment time.